### ISSUE TITLE
Alignment of Vim relative numbers

### ISSUE THREAD

**Issue by azat-io** [vim/vim#1875]
Is there any way to align current line number to right? It looks strange by default, current line is aligning to left. Why there is not option to change it?

I tried to find a solution with `:help cul`

This thing looks ugly

![vim](https://user-images.githubusercontent.com/5698350/28564313-48ea8002-7131-11e7-9d0a-2fe2f63e9479.png)

**Comment by gapeevanton** [comment]
This behavior is hardcoded. You can only modify it and recompile

**Comment by azat-io** [comment]
@gapeevanton Sounds bad. Maybe there is some instructions for solving this problem? Why is the alignment change not provided?

**Comment by chrisbra** [comment]
No there is no other possibility, except if you want to abuse the sign feature for that. Note, we are trying avoid adding options for such small things (this might be debatable here). However, there is already #680 and #1368 in which a 'numberformat' option has been whished. However, a patch which includes a test would be appreciated :)

**Comment by azat-io** [comment]
@chrisbra Thank you for the answer. I hope 'numberformat' option will be released

**Comment by cecamp** [comment]
Christian Brabandt wrote:
>
> No there is no other possibility, except if you want to abuse the sign 
> feature for that. Note, we are trying avoid adding options for such 
> small things (this might be debatable here). However, there is already 
> #680 <https://github.com/vim/vim/pull/680> and #1368 
> <https://github.com/vim/vim/issues/1368> in which a 'numberformat' 
> option has been whished. However, a patch which includes a test would 
> be appreciated :)
>
>
Before the relative numbering patch was accepted into vim's source, I 
wrote RltvNmbr (a plugin) that "abused" signs to do relative numbering.  
I noted that it too had single digit numbers left justified; it was using
   string(L)
(where L held the current linenumber).

So, I tried changing it to printf("%2d",L) .  I'm afraid that the signs 
feature does not approve of leading spaces.  So I don't think one can 
successfully abuse the sign feature for right-justified relative numbering.

Just thought I'd point that out!

Chip Campbell

**Comment by vim-ml** [comment]
On Tuesday, July 25, 2017 at 2:04:31 AM UTC-7, Azat S. wrote:
> Is there any way to align current line number to right? It looks strange by default, current line is aligning to left. Why there is not option to change it?

There is one option:

:se nonu rnu

It will right align the current line,
BUT
the current line number is then '0'
which, to me, makes more sense with 'relative' numbering

**Comment by azat-io** [comment]
@vim-ml Maybe. But I want to see exactly the number of the current line

**Comment by azat-io** [comment]
Any progress?

**Comment by chrisbra** [comment]
nobody has worked on that feature as far as i know. If you really want this feature, you could try the other issues perhaps there is a PR that implements your behaviour. We would need tests for the feature and some feedback how it works would be appreciated as well.

**Comment by Hi-Angel** [comment]
> This behavior is hardcoded. You can only modify it and recompile

Actually, I was about to report a bug, but discovered this request for an option.

Could alignment be hardcoded to right? I really don't see a reason why would it be hardcoded into an such odd position, which feels like it takes an additional space for no reason *(although aligns correctly at a thousand of lines)*. It looks as if hybrid numbers were hacked up on an UI that isn't really suited for that.

**Comment by vim-ml** [comment]
On Mon, Jul 9, 2018 at 8:33 PM, Konstantin Kharlamov
<vim-dev-github@256bit.org> wrote:
> This behavior is hardcoded. You can only modify it and recompile
>
> Actually, I was about to report a bug, but discovered this request for an
> option.
>
> Could alignment be hardcoded to right? I really don't see a reason why would
> it be hardcoded into an such odd position, which feels like it takes an
> additional space for no reason (although aligns correctly at a thousand of
> lines). It looks as if hybrid numbers were hacked up on an UI that isn't
> really suited for that.

For me having the current line number at the "odd position" makes it
easier to spot the current line. Compare:

  3 one
  2 two
  1 three
  4 four
  1 five

vs:

  3 one
  2 two
  1 three
4   four
  1 five

In the former I have to scan the numbers to see which line is the
current one, while in the latter the current line number is easy to
spot.

nazri

**Comment by tonymec** [comment]
I agree with Nazri, especially since in the default highlighting the coulours for the current and relative line numbers are not very different: both in bright yellow in a (dark) cterm, or relative numbers in brown and current line in bold brown in the (light) GUI.

The current behaviour seems intentional, see the examples at `:help number_relativenumber`.

Best regards,
Tony.

**Comment by Hi-Angel** [comment]
Okay, I respect other's workflow. The problem is that it doesn't work well even for your usecase. The behavior would make sense if it was *always* aligned, but it's not: if the file have between 100 and 999 lines, then anywhere at 100..999 it's aligned to right. Same for 1000..9999. This only doesn't hold for 1..9 and 10..99, dunno why — but then, files typically have more than 100 lines.

**Comment by lifepillar** [comment]
>The behavior would make sense if it was always aligned, but it's not

That depends on `numberwidth`. Set it to a higher value and you'll see a consistent behaviour.

**Comment by tonymec** [comment]
Also, if I'm at a "late" line in the file, it stands out by the mere fact that its high number is in the middle of one-digit numbers, e.g. 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 6543, 1, 2, 3, … OTOH, in the case of 4, 3, 2, 1, 5, 1, 2, 3, … the 5 would be much less obvious if it were aligned with the others (and, as with the default colorscheme, in the same or almost the same colour).

Best regards,
Tony.

**Comment by Hi-Angel** [comment]
I'm wondering, can we come to a compromise that doesn't look as if the feature was hacked up on a UI that requiring rewriting in the first place. How about using **bold**, *italic*, or a color to highlight the current line?

**Comment by Hi-Angel** [comment]
Another point for using a different way to highlight the current number: have anybody tried hybrid line-numbers in Arabic mode? If you open e.g. `vim -A /etc/profile` you'll find that the current line-number melds with the current line, i.e. there's no space between them.

I bet you can add right-to-left-language users to the list of unhappy ones.

**Comment by nickspoons** [comment]
@Hi-Angel The `CursorLineNr` highlight group can be used to highlight the number however you want:

```vim
:highlight CursorLineNr cterm=bold,italic gui=bold,italic
```

**Comment by Hi-Angel** [comment]
> @Hi-Angel The CursorLineNr highlight group can be used to highlight the number however you want:
> 
> ```
> :highlight CursorLineNr cterm=bold,italic gui=bold,italic
> ```

Great! Given that part is solved, are we changing the current line to align to the left, or is there anything else is blocking?

**Comment by chrisbra** [comment]
duplicate of #2204

**Comment by Hi-Angel** [comment]
@chrisbra you linked a PR. An issue can not be a duplicate of PR *(especially of one referring back to the same issue)*. An issue can be *solved by PR*, but for that PR needs to be accepted, and the one you linked haven't been.

**Comment by chrisbra** [comment]
Yes I did and I personally don't like to have PRs and issues for the same thing laying around, because it happens often enough that the corresponding PR get's closed once Bram merges it and the issue keeps laying around open and no one notices it. I personally like to have the issues in a manageable state, which would not be the case and as you can see from the issue number, we have already several hundred open issues here, which is barely manageable by the maintainers here. It does not help anybody to have this issue list here becoming a bare garbage dump site. So my decision stands. Thanks for understanding.

**Comment by wren** [comment]
Is there any chance of reviving this issue since the associated PR was abandoned?

**Comment by wren** [comment]
Also, I know this is years later and probably irrelevant by now, but did you know that putting the text "Fixes #1875" in the description of PR will automatically close that issue when the PR is merged? Makes keeping issues and PRs in sync much more manageable.

**Comment by k-takata** [comment]
Reopening, then. If someone wants this, please take over #2204.

> but did you know that putting the text "Fixes #1875" in the description of PR will automatically close that issue when the PR is merged?

Bram doesn't use the merge button, so it doesn't work.

**Comment by wren** [comment]
> Bram doesn't use the merge button, so it doesn't work.

It also works in commit messages, but fair enough. I don't want to mess with someone's work flow. 

Thanks for reopening!

**Comment by askfiy** [comment]
Well, I think this function should be necessary

**Comment by felixakiragreen** [comment]
please 🙏 how do we make this happen?

**Comment by Hakkadaikon** [comment]
Since this issue was last closed after a pull request was submitted, several years ago, I think it's okay to close it as "not planned" for now

**Comment by chrisbra** [comment]
thanks, closing as not planned
